src/base/utils/fs.cpp

```diff
@@ -49,17 +45,58 @@
 #include <unistd.h>
 #endif
 
+#include <QByteArray>
 #include <QDebug>
 #include <QDir>
 #include <QDirIterator>
 #include <QFile>
 #include <QFileInfo>
+#include <QIODevice>
 #include <QStorageInfo>
 #include <QRegularExpression>
 
 #include "base/bittorrent/torrenthandle.h"
 #include "base/global.h"
 
+Utils::Fs::FileOutputIterator::FileOutputIterator(QIODevice &device, const int bufferSize)
+    : m_device {&device}
+    , m_buffer {std::make_shared<QByteArray>()}
+    , m_bufferSize {bufferSize}
+{
+    m_buffer->reserve(bufferSize);
+}
+
+Utils::Fs::FileOutputIterator::~FileOutputIterator()
+{
+    m_device->write(*m_buffer);
+    m_buffer->clear();
+}
+
+Utils::Fs::FileOutputIterator &Utils::Fs::FileOutputIterator::operator=(const char c)
+{
+    m_buffer->append(c);
+    if (m_buffer->size() >= m_bufferSize) {
+        m_device->write(*m_buffer);
```

Comment:
You should provide error handling (throw an exception) here and in destructor, shouldn't you?

Reply:
Added error handling here, but I'm a bit hesitant to add it to destructor, seems dangerous to me,

Reply:
>but I'm a bit hesitant to add it to destructor, seems dangerous to me,

Then possible error at last write will be unhandled. Isn't it dangerous?

Reply:
Well, seems throwing in destructor is really dangerous...
Then we shouldn't throw in this class at all to keep error handling consistent. Seems we can just check for QFile error status after all.

Reply:
>Seems we can just check for QFile error status after all.

I'm not sure how that would be useful. As continuous invoking `m_device->write()` the previous error will be overwrite by the later one and there is one last invoke in destructor.

Reply:
How can we handle errors in a reliable way?

Reply:
We can have a `std::shared<bool> m_hasError;` field that would record `true` if any error occurs and users of this class can inspect it (only after `bencode()` ends, I think it is not so useful), besides the class doesn't look like an iterator anymore...

Reply:
Maybe we should fallback to no-op in case of errored device? Then we can check its status at the end as I suggested before.
You don't think we can just ignore it generally, do you?

Reply:
>Maybe we should fallback to no-op in case of errored device? 

Done.